template.tpl

```diff
@@ -0,0 +1,581 @@
+    "macrosInSelect": false,
+    "selectItems": [
+      {
+        "value": "matomo.js",
+        "displayValue": "matomo.js"
+      },
+      {
+        "value": "piwik.js",
+        "displayValue": "piwik.js"
+      },
+      {
+        "value": "js/",
+        "displayValue": "js/"
+      },
+      {
+        "value": "js/tracker.php",
+        "displayValue": "js/tracker.php"
+      }
+    ],
+    "simpleValueType": true,
+    "help": "Here you can configure the source path of the Matomo Tracker JavaScript, if you are not using the \"Bundle Tracker\" option.  Default: matomo.js"
+  },
+  {
+    "type": "SELECT",
+    "name": "trackingEndpoint",
+    "displayName": "Tracking Request Target Path",
+    "macrosInSelect": false,
+    "selectItems": [
+      {
+        "value": "matomo.php",
+        "displayValue": "matomo.php"
+      },
+      {
+        "value": "piwik.php",
+        "displayValue": "piwik.php"
+      },
+      {
+        "value": "js/",
+        "displayValue": "js/"
+      },
+      {
+        "value": "js/tracker.php",
+        "displayValue": "js/tracker.php"
+      }
+    ],
+    "simpleValueType": true,
+    "help": "Here you can configure the target path for tracking requests.  Default: matomo.php"
+  }
+]
+
+
+___SANDBOXED_JS_FOR_WEB_TEMPLATE___
+
+// Enter your template code here.
+const log = require('logToConsole');
+const createQueue = require('createQueue');
+const injectScript = require('injectScript');
+log('data =', data);
+
+const onSuccess = () => {
+  log('Tracker.js loaded successfully');
+  data.gtmOnSuccess();
+};
+
+const onFailure = () => {
+  log('Tracker.js loading failed');
+  data.gtmOnFailure();
+};
+
+if (data.matomoSubDomain && data.idSite) {
+  const injectScript = require('injectScript');
+  let baseUrl = 'https://cdn.matomo.cloud/' + data.matomoSubDomain + data.matomoDomain  + '/';
+  
+  // if you want to test this on your local instance, uncomment the below line add your local instance url and add the same inside the permissions tab in inject scripts
+  // baseUrl='{YOURMATOMO_URL}';
+ 
+  const url = baseUrl + data.jsEndpoint;
+  log(url);
+  const _paq = createQueue('_paq');
+  _paq(['setSiteId',data.idSite]);
+  _paq(['setTrackerUrl',baseUrl + data.trackingEndpoint]);
+  
+  let disableConfigValues = {'disableTrackPageview':'trackPageView', 'disableLinkTracking':'enableLinkTracking', 'disableCrossDomainLinking':'enableCrossDomainLinking'};
+  paqDisable(_paq, disableConfigValues);
+  
+  let enableConfigValues = {'enableDoNotTrack':'setDoNotTrack', 'enableJSErrorTracking':'enableJSErrorTracking', 'enableHeartBeatTimer':'enableHeartBeatTimer','trackAllContentImpressions':'trackAllContentImpressions', 'trackVisibleContentImpressions':'trackVisibleContentImpressions','disableCookies':'disableCookies','requireConsent':'requireConsent', 'requireCookieConsent':'requireCookieConsent','setSecureCookie':'setSecureCookie'};
+  paqEnable(_paq, enableConfigValues);
```

Comment:
I'm not sure if all these methods wouldn't be triggered in the right order as they are defined in the regular JS tracking code? 
<img width="964" alt="image" src="https://user-images.githubusercontent.com/273120/175926715-f738db7e-cb27-4c90-80b2-b36013a201f7.png">

Or maybe it doesn't matter though in case we are using `window._paq.push(...)`.

Reply:
I updated the position of execution as per `applyFirst`, please check